### PR TITLE
CreateImageWizard: Azure upload step

### PR DESCRIPTION
**Pull request by Gundersanne**
image

---

request generated (bogus values):
```json
{
  "distribution": "rhel-8",
  "image_requests": [
    {
      "architecture": "x86_64",
      "image_type": "vhd",
      "upload_requests": [
        {
          "type": "azure",
          "options": {
            "tenant_id": "oqiwjd",
            "subscription_id": "qowidj",
            "resource_group": "qowidj"
          }
        }
      ]
    }
  ],
  "customizations": {
    "subscription": {
      "organization": 000000
    }
  }
}
```

### CODE REVIEW

File: src/SmartComponents/CreateImageWizard/CreateImageWizard.js [hunk 1]
```diff
@@ -127,6 +131,29 @@ class CreateImageWizard extends Component {
...
+                { label: 'Azure destination tenant ID', value: 'A tenant ID is required' };
+        }
+
+        let subscription = this.state.uploadAzure.options.destination.subscription;
+        if (subscription === null || subscription === '') {
+            uploadAzureErrors['azure-destination-subscription'] =
+                { label: 'Azure destination subscription', value: 'A subscription is required' };
+        }
+
+        let resource_group = this.state.uploadAzure.options.destination.resource_group;
```
Comment (jkozol) [inline src/SmartComponents/CreateImageWizard/CreateImageWizard.js:149]: Can you switch `.options.destination.` to just `.options`?

File: src/PresentationalComponents/CreateImageWizard/WizardStepUploadAzure.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                </Text>
+
+                <a href="https://login.microsoftonline.com/common/oauth2/authorize?client_id=b94bb246-b02c-4985-9c22-d44e66f657f4
+&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient">
+            Authorize Image Builder on Azure <ExternalLinkAltIcon />
+                </a>
+                <Title headingLevel="h4">Permission status</Title>
+                { (props.uploadAzure.azureAuthorized && <><ResourcesEmptyIcon /> Image Builder is not authorized</>) ||
+              <><CheckCircleIcon /> Image Builder is authorized</> }<br />
+                { (props.uploadAzure.azureRoleGranted &&
```
Comment (jkozol) [inline src/PresentationalComponents/CreateImageWizard/WizardStepUploadAzure.js:33]: Can you set default values for `azureAuthorized` and `azureRoleGranted` in the state?

**Review by jkozol — COMMENTED** [review]
Nice start on this! Thanks!!

File: src/PresentationalComponents/CreateImageWizard/WizardStepUploadAzure.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            Image Builder must be authorized by an account owner.<br />
+                    <a href="https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow">
+                        <small>Learn more</small></a>
+                </Text>
+
+                <a href="https://login.microsoftonline.com/common/oauth2/authorize?client_id=b94bb246-b02c-4985-9c22-d44e66f657f4
+&redirect_uri=https%3A%2F%2Flogin.microsoftonline.com%2Fcommon%2Foauth2%2Fnativeclient">
+            Authorize Image Builder on Azure <ExternalLinkAltIcon />
+                </a>
+                <Title headingLevel="h4">Permission status</Title>
```
Comment (ondrejbudai) [inline src/PresentationalComponents/CreateImageWizard/WizardStepUploadAzure.js:30]: With the current MVP way of authorizing our Azure application, we can't check the permission status. So maybe let's hide this for now?

File: src/PresentationalComponents/CreateImageWizard/WizardStepUploadAzure.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                <FormGroup isRequired label="Tenant ID" fieldId="azure-destination-tenant-id"
+                    helperTextInvalid={ (props.errors['azure-destination-tenant-id'] && props.errors['azure-destination-tenant-id'].value) || '' }
+                    validated={ (props.errors['azure-destination-tenant-id'] && 'error') || 'default' }>
+                    <TextInput value={ props.uploadAzure.options.tenant_id || '' }
+                        type="text" aria-label="Azure destination tenant-id" id="azure-destination-tenant-id"
+                        data-testid="azure-destination-tenant-id" isRequired
+                        onChange={ value =>
+                            props.setUploadOptions('azure', Object.assign(props.uploadAzure.options, { tenant_id: value })) } />
+                </FormGroup>
+                <FormGroup isRequired label="Subscription" fieldId="azure-destination-subscription"
```
Comment (ondrejbudai) [inline src/PresentationalComponents/CreateImageWizard/WizardStepUploadAzure.js:52]:
```suggestion
                <FormGroup isRequired label="Subscription ID" fieldId="azure-destination-subscription"
```

Sadly, we need a subscription ID here.

**Review by ondrejbudai — COMMENTED** [review]
Looks awesome! Few suggestions inline.

**Comment by ondrejbudai** [comment]
Looks good!

**Comment by teg** [comment]
Commit message seems wrong, but otherwise: really nice!

**Review by teg — APPROVED** [review]

**Pull request merged**